gateway-ha/src/main/java/io/trino/gateway/proxyserver/RouteToBackendResource.java

```diff
@@ -85,4 +86,14 @@ public void deleteHandler(
         String remoteUri = routingTargetHandler.getRoutingDestination(servletRequest);
         proxyRequestHandler.deleteRequest(servletRequest, asyncResponse, URI.create(remoteUri));
     }
+
+    @PUT
+    public void putHandler(
+            String body,
+            @Context HttpServletRequest servletRequest,
+            @Suspended AsyncResponse asyncResponse)
+    {
+        String remoteUri = routingTargetHandler.getRoutingDestination(servletRequest);
```

Comment:
create a `MultiReadHttpServletRequest` from the `servletRequest` as in the `postHandler` method. 

Reading the body of a standard `HttpServletRequest` is a destructive operation, using the `MultiReadHttpServletRequest` ensures that nothing in the routing logic alters the request that we forward.